code/modules/power/antimatter/shielding.dm

```diff
@@ -79,11 +84,21 @@ proc/cardinalrange(var/center)
 		for(var/obj/machinery/power/am_control_unit/AMC in cardinalrange(src))
 			if(AMC.add_shielding(src))
 				break
-		if(!priorscan)
-			sleep(20)
-			controllerscan(1)//Last chance
-			return
-		qdel(src)
+		if(!mapped) // Prevent rescanning and suicide if it's part of the map
+			if(!priorscan)
+				sleep(20)
+				controllerscan(1)//Last chance
+				return
+			qdel(src)
+
+// Find surrounding unconnected shielding and add them to our controller
+/obj/machinery/am_shielding/proc/assimilate()
```

Comment:
Reading through the code I'm not sure this is the best way to do this, or rather, it appears to be kind of tacked on to the present functionality